### DISCUSSION TITLE
making thrust run boost::geometry code ?

### DISCUSSION THREAD

**Discussion by mkandulavm** [NVIDIA/cccl#1054] [Thrust]
Hi

What is the best way to try running  code that utilizes boost::geometry  and its associated ::union_   code run on GPU with thrust ?

A sample to merge a test geometry using thrust will be appreciated !

Thanks.

**Comment by alliepiper** [comment]
I'm not familiar with that boost module, but I suspect that you'll have problems. Boost doesn't provide gpu-compatible implementations of their algorithms, and since they lack host/device markup they won't be able to be used from device contexts with CUDA compilers.

I don't think this is possible, but I'll leave this open in case someone else has some ideas.

**Comment by mkandulavm** [comment]
@allisonvacanti thanks for the reply. 
Let me know if you can suggest any alternative approaches.  My requirement is to merge 2D line strings and polygons.

**Comment by jrhemstad** [comment]
You might checkout https://github.com/rapidsai/cuspatial, it has some geometry algorithms. @cwharris should be able to answer any questions for you.
